RealmSwift/Combine.swift

```diff
@@ -459,6 +459,35 @@ extension AnyRealmCollection: RealmSubscribable {
     }
 }
 
+/// A subscription which wraps a Realm AsyncOpenTask.
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, *)
+@frozen public struct AsyncOpenSubscription: Subscription {
+    private let task: Realm.AsyncOpenTask
+
+    internal init(task: Realm.AsyncOpenTask,
+                  callbackQueue: DispatchQueue,
+                  onProgressNotificationCallback: ((SyncSession.Progress) -> Void)?) {
+        self.task = task
+        self.task.addProgressNotification(queue: callbackQueue, block: onProgressNotificationCallback ?? { _ in })
```

Comment:
This should instead skip calling addProgressNotification if no callback is supplied.